src/atlantis/router/backend/server.go

```diff
@@ -78,9 +78,10 @@ func (s *Server) Handle(logRecord *logger.HAProxyLogRecord, tout time.Duration)
 	logRecord.UpdateTr(tstart, tend)
 	select {
 	case resErr := <-resErrCh:
-		if resErr.Error == nil {
+		if resErr.Response != nil {
 			defer resErr.Response.Body.Close()
-
```

Comment:
Just a question (since I am not familiar with router code) resErr.Error == null means there is no error and we do not need to close connection in this case?